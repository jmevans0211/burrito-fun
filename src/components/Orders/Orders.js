import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setOrders } from '../../actions';
import { getOrders, deleteOrder } from '../../apiCalls';
import './Orders.css';

export class Orders extends Component {
  constructor() {
    super();
    this.state = {
    }
  }
  componentDidMount() {
    const {setOrders} = this.props
    getOrders()
    .then(data => setOrders(data.orders))
    .catch(err => console.error('Error fetching:', err));
  }
  
handleDelete(e) {
  const id = e.target.id
  deleteOrder(id)
}

  render() {
    const {orders} = this.props
    const orderEls = orders.map(order => {
      return (
        <div key={order.id} className="order">
          <h3>{order.name}</h3>
          <ul className="ingredient-list">
            {order.ingredients.map(ingredient => {
              return <li key={ingredient}>{ingredient}</li>
            })}
          </ul>
          <button id={order.id} type="button" onClick={(e) => this.handleDelete(e)}>Delete Order</button>
        </div>
      )
    });
    return (
      <section>
        { orderEls.length ? orderEls : <p>No orders yet!</p> }
      </section>
    )
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setOrders,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Orders);