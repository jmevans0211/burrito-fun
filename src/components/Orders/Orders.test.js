import React from 'react';
import { shallow } from 'enzyme';
import { Orders, mapStateToProps, mapDispatchToProps } from './Orders';

describe('Orders', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Orders orders = {
      [
        {
          id : 1,
          name : 'Jeannie',
          ingredients : ['lettuce', 'tomato', 'onion']
        }
      ]
    } />);
  });

  it('should match snapshot', () => {
  expect(wrapper).toMatchSnapshot();
  });


}); //<---- end of ORDERS describe block

describe('mapStateToProps', () => {
  it('should return an object with an orders array', () => {
    const mockState = {
      orders: [
        {
          id: 7,
          name: "Princess Buttercup",
          ingredients: ['salt', 'n', 'peppa']
        },
        {
          id: 8,
          name: "Princess GumDrop",
          ingredients: ['suga', 'n', 'spyce']
        }
      ],
      filter: 'SET_ORDERS'
    };
    const expected = {
      orders: mockState.orders
    }

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });


}); // <---- end of mapSTATEtoProps describe block


describe('mapDispatchToProps', () => {


}); //<<--- end of mapDISPATCHtoPRops describe block