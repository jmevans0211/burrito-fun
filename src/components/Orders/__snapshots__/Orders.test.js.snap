// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match snapshot 1`] = `
<section>
  <div
    className="order"
    key="1"
  >
    <h3>
      Jeannie
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="lettuce"
      >
        lettuce
      </li>
      <li
        key="tomato"
      >
        tomato
      </li>
      <li
        key="onion"
      >
        onion
      </li>
    </ul>
  </div>
</section>
`;
